liwords-ui/src/settings/preferences.tsx

```diff
@@ -185,11 +193,90 @@ export const Preferences = React.memo((props: Props) => {
     setUserBoard(boardStyle);
   }, []);
 
+  const [reevaluateTileOrderOptions, setReevaluateTileOrderOptions] = useState(
+    0
+  );
   const [tileOrder, setTileOrder] = useState(preferredSortOrder ?? '');
-  const handleTileOrderChange = useCallback((value) => {
-    setTileOrder(value);
-    setPreferredSortOrder(value);
+  const handleTileOrderAndAutoShuffleChange = useCallback((value) => {
+    try {
+      const parsedStuff = JSON.parse(value);
+      const {
+        tileOrder: newTileOrder,
+        autoShuffle: newAutoShuffle,
+      } = parsedStuff;
+      setTileOrder(newTileOrder);
+      setPreferredSortOrder(newTileOrder);
+      setSharedEnableAutoShuffle(newAutoShuffle);
+      setReevaluateTileOrderOptions((x) => (x + 1) | 0);
+    } catch (e) {
+      console.error(e);
+    }
   }, []);
+  const localEnableAutoShuffle = sharedEnableAutoShuffle;
+  const tileOrderValue = React.useMemo(
+    () => makeTileOrderValue(tileOrder, localEnableAutoShuffle),
+    [tileOrder, localEnableAutoShuffle]
+  );
+  const tileOrderOptions = React.useMemo(() => {
+    void reevaluateTileOrderOptions;
+    const ret: Array<{ name: React.ReactElement; value: string }> = [];
+    const pushTileOrder = (
+      name: string,
+      value: string,
+      autoShuffle: boolean
+    ) => {
+      // Design only wants "Random" for "Alphabetical".
+      if (
+        !(
+          (tileOrder === value && autoShuffle === localEnableAutoShuffle) ||
+          name === 'Alphabetical' ||
+          !autoShuffle
```

Comment:
I'm not sure what this does - does it only add the `Random` option in some cases?